modules/renter/download.go

```diff
@@ -115,6 +115,14 @@ package renter
 // Overall I don't think it's going to be all that difficult, but it's not
 // nearly as clean-cut as some of the other potential extensions that we can do.
 
+// TODO: Right now the whole download will build and send off chunks even if
+// there are not enough hosts to download the file, and even if there are not
+// enough hosts to download a particular chunk. For the downloads and chunks
+// which are doomed from the outset, we can skip some computation by checking
+// and failing earlier. Another optimization we can make is to not count a
+// worker for a chunk if the worker's contract does not appear in the chunk
+// heap.
+
```

Comment:
Should that be part of this PR? Feels like a step back to not have it from the start. Otherwise we waste money on bandwidth.

Reply:
The workers check whether a chunk is viable before they start downloading. Also, the general operating assumption is that all files are able to be downloaded.